src/app/country/country.component.ts

```diff
@@ -1,42 +1,40 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from "@angular/router";
-import { WorldHttpService } from "../world-http.service";
-import { Location } from "@angular/common";
-import { AppInterface } from "../../app/application-interface";
+import { ActivatedRoute, Router } from '@angular/router';
+import { WorldHttpService } from '../world-http.service';
+import { Location } from '@angular/common';
+import { AppInterface } from '../../app/application-interface';
 import { Subscription } from 'rxjs';
 import { CountriesDTO } from '../dtos/countries-dto';
 
 @Component({
   selector: 'app-country',
-  templateUrl:'./country.component.html',
+  templateUrl: './country.component.html',
   styleUrls: ['./country.component.css'],
-  providers: [Location]
+  providers: [Location],
 })
 export class CountryComponent implements OnInit {
-
-
-  public countries: CountriesDTO[]=[];
-  public returnParameters!: Subscription|undefined;
+  public countries: AppInterface[] = [];
```

Comment:
@Masian-Parkire  You were trying to assign values of type AppInterface to contries which had the type CountriesDTO causing clashes with value assignment